scripts/dev.sh

```diff
@@ -0,0 +1,246 @@
+
+#   Available commands:
+
+#   init      - Initialize new OCP cluster by patching registry settings and logging in
+#   build     - Build and push all images
+#   all       - Run all targets
+#   catalog   - Apply CatalogSource (install operator into operator hub)
+#   subscribe - Apply Subscription (install operator onto cluster)
+
+set -Eeo pipefail
+
+readonly USAGE="Usage: dev.sh all | init | build | catalog | subscribe  [ -host <ocp registry hostname url> -version <operator verion to build> -image <image name> -bundle <bundle image> -catalog <catalog image> -name <operator name> -namespace <namespace> -tempdir <temp dir> ]"
+
+main() {
+
+  parse_args "$@"
+
+  if [[ -z "${COMMAND}" ]]; then
+    echo
+    echo "${USAGE}"
+    echo
+    exit 1
+  fi
+  
+  if [[ -z "${OCP_REGISTRY_URL}" ]]; then
+    echo
+    echo "OCP registry hostname is required. Set CLUSTER_REGISTRY_URL or use the -host command line option."
+    echo
+    echo "${USAGE}"
+    exit 1
+  fi
+
+  # Set defaults unless overridden. 
+  NAMESPACE=${NAMESPACE:="websphere-liberty"}
+  VERSION=${VERSION:="0.0.1"}
+  VVERSION=${VVERSION:=v$VERSION}
+  OPERATOR_NAME=${OPERATOR_NAME:="operator"}
+  IMAGE_TAG_BASE=${IMAGE_TAG_BASE:=$OCP_REGISTRY_URL/$NAMESPACE/$OPERATOR_NAME:$VVERSION}
+  IMG=${IMG:=$OCP_REGISTRY_URL/$NAMESPACE/$OPERATOR_NAME:$VVERSION}
+  BUNDLE_IMG=${BUNDLE_IMG:=$OCP_REGISTRY_URL/$NAMESPACE/$OPERATOR_NAME-bundle:$VVERSION}
+  CATALOG_IMG=${CATALOG_IMG:=$OCP_REGISTRY_URL/$NAMESPACE/$OPERATOR_NAME-catalog:$VVERSION}
+  MAKEFILE_DIR=${MAKEFILE_DIR:=..}
+  TEMP_DIR=${TEMP_DIR:=/tmp}
+  
+  if [[ "$COMMAND" == "all" ]]; then
+     init_cluster
+     login_registry
+     build
+     apply_catalog
+     apply_subscribe
+  elif [[ "$COMMAND" == "init" ]]; then
+     init_cluster
+     login_registry
+  elif [[ "$COMMAND" == "build" ]]; then
+     build
+     bundle
+     catalog
+  elif [[ "$COMMAND" == "catalog" ]]; then
+     apply_catalog
+  elif [[ "$COMMAND" == "subscribe" ]]; then
+     apply_og
+     apply_subscribe
+  else 
+    echo
+    echo "Command $COMMAND unrecognized."
+    echo
+    echo "${USAGE}"
+    exit 1
+  fi
+
+}
+
+#############################################################################
+# Setup an OCP cluster to use the private registry, insecurely (testing only)
+#############################################################################
+init_cluster() {
+    oc patch image.config.openshift.io/cluster  --patch '{"spec":{"registrySources":{"insecureRegistries":["'$OCP_REGISTRY_URL'"]}}}' --type=merge
+    oc patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
+    oc project $NAMESPACE > /dev/null 2>&1
+    if [[ $? -ne 0 ]]; then
+      oc new-project $NAMESPACE 
+    fi
+}
+
+login_registry() {
+    HOST=$(oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
+    podman login -u kubeadmin -p $(oc whoami -t) --tls-verify=false $HOST  
```

Comment:
I had to change `podman` to `docker` and remove `--tls-verify=false`

Reply:
I updated the script to work with docker or podman.